network/p2p/p2p_setup.go

```diff
@@ -115,6 +117,52 @@ func (n *p2pNetwork) IsBadPeer(logger *zap.Logger, peerID peer.ID) bool {
 	return n.idx.IsBad(logger, peerID)
 }
 
+func (n *p2pNetwork) canAcceptNewConnection(lg *zap.Logger, peerID peer.ID) bool {
+	if n.idx == nil {
+		return false
+	}
+
+	maxPeers := n.cfg.MaxPeers
+	inBoundLimit := int(float64(n.cfg.MaxPeers) * inboundRatio)
```

Comment:
should probably be `int(float64(n.cfg.MaxPeers) * inboundRatio)` -> `int(float64(maxPeers) * inboundRatio)`